vizro-ai/tests/integration/test_dashboard.py

```diff
@@ -55,97 +171,22 @@ def test_simple_dashboard(dash_duo, model_name):  # noqa: PLR0915
-    except IndexError:
-        grid_2 = 0
-
-    try:
-        components_num_1 = [1 if len(dashboard.pages[0].components) == 1 else 0][0]
-    except IndexError:
-        components_num_1 = 0
-    try:
-        components_num_2 = [1 if len(dashboard.pages[1].components) == 3 else 0][0]
-    except IndexError:
-        components_num_2 = 0
-
-    try:
-        controls_num_1 = [1 if len(dashboard.pages[0].controls) == 0 else 0][0]
-    except IndexError:
-        controls_num_1 = 0
-    try:
-        controls_num_2 = [1 if len(dashboard.pages[1].controls) == 2 else 0][0]
-    except IndexError:
-        controls_num_2 = 0
-
-    try:
-        comps_1 = [components.type for components in dashboard.pages[0].components]
-        components_types_1 = [1 if comps_1.count("ag_grid") == 1 else 0][0]
-    except IndexError:
-        components_types_1 = 0
-    try:
-        comps_2 = [components.type for components in dashboard.pages[1].components]
-        components_types_2 = [1 if comps_2.count("card") == 2 and comps_2.count("graph") == 1 else 0][0]
-    except IndexError:
-        components_types_2 = 0
-
-    try:
-        cntrls_2 = [controls.type for controls in dashboard.pages[1].controls]
-        controls_types_2 = [1 if cntrls_2.count("filter") == 2 else 0][0]
-    except IndexError:
-        controls_types_2 = 0
-
-    prescore = [
-        pages,
-        pages_num,
-        grid_1,
-        grid_2,
-        components_num_1,
-        components_num_2,
-        controls_num_1,
-        controls_num_2,
-        components_types_1,
-        components_types_2,
-        controls_types_2,
-    ]
-    score = sum(prescore)
-
-    with open(f"{report_dir}/report_model_{model_name}.csv", "a", newline="") as csvfile:
-        writer = csv.writer(csvfile, delimiter=",")
-        writer.writerow([f"Vizro type = {vizro_type}, Datetime = {datetime.now()}"])
-        writer.writerow([])
-        writer.writerow(["Description, Score"])
-        writer.writerow([f"Pages exists, {pages}"])
-        writer.writerow([f"Correct pages number, {pages_num}"])
-        writer.writerow([f"Correct grid first page, {grid_1}"])
-        writer.writerow([f"Correct grid second page, {grid_2}"])
-        writer.writerow([f"Correct components number first page, {components_num_1}"])
-        writer.writerow([f"Correct components number second page, {components_num_2}"])
-        writer.writerow([f"Correct controls number first page, {controls_num_1}"])
-        writer.writerow([f"Correct controls number second page, {controls_num_2}"])
-        writer.writerow([f"Correct components types first page, {components_types_1}"])
-        writer.writerow([f"Correct components types second page, {components_types_2}"])
-        writer.writerow([f"Correct controls types second page, {controls_types_2}"])
-        writer.writerow([f"Total, {(score / len(prescore)):.4f}"])
-        writer.writerow([])
-        writer.writerow([])
+    logic(
```

Comment:
Indeed this is really cool, and I assume that this could be extended then easily by adding new prompts and more complicated checking logic?

Maybe one suggestion, could we make the logic even easier to setup, e.g

```
components = [{"page_num": 1, "type": "ag_grid"},{"page_num": 2, "type": "card"},...]
controls = ...
layout ... (in future)
```
and derive from that that there are x pages, and on page 1 should be this, on page 2 should be this. We could even work with dataclasses or something to streamline the config a bit

Reply:
this type of config is not showing how many `cards` we have on the page or `ag_grid`. my thought was that every option will have a score, so when you creating your config you will know what you'll get in the report.
but let's discuss it on Monday.

Reply:
Let's quickly chat